mpitest/test_mpi_matxmat.py

```diff
@@ -108,18 +109,24 @@ def test_fan_in_parallel_sets(self):
 
         prob.driver.parallel_derivs(['p1.x1','p2.x2'])
 
+        if MPI:
+            expected = [('p1.x1','p2.x2'),('p3.x3',)]
+        else:
+            expected = [('p1.x1',),('p2.x2',),('p3.x3',)]
```

Comment:
why should we return anything differently here, if we're not under MPI? We can't solve them in parallel, but we could at least return them in the grouped style. 

Also, I'm assuming that in serial you don't allocate the multiple rhs's?